CMakeLists.txt

```diff
@@ -78,7 +78,14 @@ Generating build files for OpenShot
 #### Enable C++11 (for JUCE)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+
+#### Optional C++ restirctions
+option(USE_CXX_GNU_EXTENSIONS "Allow GNU C++ extensions" OFF)
+if(USE_CXX_GNU_EXTENSIONS)
+	set(CMAKE_CXX_EXTENSIONS ON)
+else()
+	set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
```

Comment:
Instead of doing all this, I would just make the standard variable configurable (which it already is, technically). You can replace the whole thing with just:

```cmake
set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "Allow compiler C++ language extensions")
```

Because there's no "FORCE" there, this will show up as a configurable build parameter and if the user sets it pre-generation step (on either the command line or in `cmake-gui`), i'll override the default "OFF".